packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -314,50 +313,50 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
 
       const startIcon = startIconProp ?? icon;
 
-      const menuItem: JSX.Element = itemRenderer ? (
+      return itemRenderer ? (
         itemRenderer(option, { close: () => setIsOpen(false), isSelected })
       ) : (
-        <MenuItem startIcon={startIcon}>{label}</MenuItem>
+        <MenuItem
+          startIcon={startIcon}
+          key={`${label}-${index}`}
+          isSelected={isSelected}
+          onClick={() => {
+            if (option.disabled) {
+              return;
+            }
+            if (isSingleOnChange(onChange, multiple)) {
+              onChange?.(option.value);
+              setIsOpen(false);
+            } else {
+              onChange?.(option.value, isSelected ? 'removed' : 'added');
+            }
+
+            // update live region
+            if (isMultipleEnabled(value, multiple)) {
+              const prevSelectedValue = value || [];
+              const newSelectedValue = isSelected
+                ? prevSelectedValue.filter((i) => option.value !== i)
+                : [...prevSelectedValue, option.value];
+              setLiveRegionSelection(
+                options
+                  .filter((i) => newSelectedValue.includes(i.value))
+                  .map((item) => item.label)
+                  .filter(Boolean)
+                  .join(', '),
+              );
+            }
+          }}
+          ref={(el: HTMLLIElement) => {
+            if (isSelected && !multiple) {
+              el?.scrollIntoView({ block: 'nearest' });
+            }
+          }}
+          role='option'
+          {...restOption}
+        >
+          {label}
+        </MenuItem>
       );
-
-      return React.cloneElement<MenuItemProps>(menuItem, {
```

Comment:
the original reason for using `cloneElement` here was for adding extra stuff to menuItem returned by custom `itemRenderer`.

so if we're removing it, there needs to be a way to expose equivalent functionality

Reply:
I was thinking if we need to remove this cloneElement statement?
Reverted changes for now.

Reply:
it's ok to keep for now because we are not passing `className` inside it. we can try to figure out a better solution in the future